hoomd/mpcd/RejectionVirtualParticleFiller.h

```diff
@@ -0,0 +1,175 @@
+                                       Scalar density,
+                                       unsigned int type,
+                                       std::shared_ptr<::Variant> T,
+                                       unsigned int seed,
+                                       std::shared_ptr<const Geometry> geom)
+        : mpcd::VirtualParticleFiller(sysdata, density, type, T, seed), m_geom(geom)
+        {
+        m_exec_conf->msg->notice(5) << "constructing MPCD RejectionVirtualParticleFiller" << std::endl;
+        }
+
+        //! Destructor
+        ~RejectionVirtualParticleFiller()
+        {
+        m_exec_conf->msg->notice(5) << "Destroying MPCD RejectionVirtualParticleFiller" << std::endl;
+        }
+
+        //! Get the streaming geometry
+        std::shared_ptr<const Geometry> getGeometry() const
+            {
+            return m_geom;
+            }
+
+        //! Set the streaming geometry
+        void setGeometry(std::shared_ptr<const Geometry> geom)
+            {
+            m_geom = geom;
+            }
+
+        //! Fill the particles outside the confinement
+        void fill(unsigned int timestep);
+
+    protected:
+        std::shared_ptr<const Geometry> m_geom;
+
+    };
+
+
+template<class Geometry>
+void RejectionVirtualParticleFiller<Geometry>::fill(unsigned int timestep)
+    {
+    // Number of particles that we need to draw (constant)
+    const BoxDim& box = m_pdata->getBox();
+    const Scalar3 lo = box.getLo();
+    const Scalar3 hi = box.getHi();
+    const unsigned int m_NVirtMax = round(m_density*box.getVolume());
+
+    // Step 1: Create temporary GPUArrays to draw Particles locally using the worst case estimate for number
+    //         number of particles.
+    GPUArray<Scalar4> pos_loc(m_NVirtMax, m_exec_conf);
+    ArrayHandle<Scalar4> m_pos_loc(pos_loc, access_location::host, access_mode::readwrite);
+
+    // Step 2: Draw the particles.
+    unsigned int pidx = 0;
+    for (unsigned int i=0; i < m_NVirtMax; ++i)
+        {
+        // TODO: Currently just using the constant for SlitGeometryFiller which needs to be changed
+        // NOTE: particle tags neglected for the RNG here since we don't know them.
+        hoomd::RandomGenerator rng(hoomd::RNGIdentifier::SlitGeometryFiller, m_seed, timestep);
+
+        Scalar3 particle = make_scalar3(hoomd::UniformDistribution<Scalar>(lo.x, hi.x)(rng),
+                                       hoomd::UniformDistribution<Scalar>(lo.y, hi.y)(rng),
+                                       hoomd::UniformDistribution<Scalar>(lo.z, hi.z)(rng));
+
+        if (m_geom->isOutside(particle))
+            {
+            m_pos_loc.data[pidx] = make_scalar4(particle.x,
+                                              particle.y,
+                                              particle.z,
+                                              __int_as_scalar(m_type));
+            pidx += 1;
```

Comment:
The idiomatic way to write this compactly is:
```suggestion
            m_pos_loc.data[pidx++] = make_scalar4(particle.x,
                                              particle.y,
                                              particle.z,
                                              __int_as_scalar(m_type));
```